src/peakrdl_python/templates/peakrdl_python/memory.py

```diff
@@ -4,13 +4,14 @@
 """
 from array import array as Array
 from typing import List
+from abc import ABC
 
-from .base import Node
+from .base import Node, AddressMap
 
 from .callbacks import CallbackSet
 
 
-class Memory(Node):
+class Memory(Node, ABC):
```

Comment:
Now that I have fixed the bug that memories that allows memories to have registers within them. I think the 

```{{ child_getter(node, "get_registers",   "Reg",        systemrdlRegNode) }}`` is needed in here

Reply:
I have made a separate `readable_registers` and `writeable_registers` getter.

Reply:
I have moved the readable and writeable register to `reg_definition.py.jinja` to allow its use in the memories